iceoryx_hoofs/test/moduletests/test_posix_mutex.cpp

```diff
@@ -134,4 +134,39 @@ TEST_F(Mutex_test, TryLockWithRecursiveMutexReturnsFalseWhenMutexLockedInOtherTh
     sutRecursive.unlock();
     lockThread.join();
 }
+
+TEST_F(Mutex_test, LockedMutexBlocks)
+{
+    std::atomic_bool isLockFinished{false};
+    sutNonRecursive.lock();
+
+    std::thread lockThread([&] {
```

Comment:
Would be good to have some timeout in this thread to avoid a deadlock in these tests. Maybe we can use the `timedwait` from the semaphore here?

Reply:
I will release our Watchdog :)

Reply:
There is already a watchdog setup in the `SetUp` method of the test (line 37) which will terminate the test if it deadlocks.

Reply:
It will terminate the whole executable as far as I can see. There may be no other way, but consider trying what I wrote in the comment at line 37.